ocs_ci/utility/kms.py

```diff
@@ -2595,3 +2543,86 @@ def set_kms_endpoint(address):
     params = f'{{"data": {{"{addr_attribute}": "{address}"}}}}'
     ocs_kms_configmap.patch(params=params, format_type="merge")
     return ocs_kms_configmap.get().get("data")[addr_attribute]
+
+
+def fetch_osd_secret_from_vault(device_handle, vault_backend_path):
+    """Fetch the OSD encryption key for the given device handle from Vault.
+
+    Args:
+        device_handle (str): The device handle for which to retrieve the OSD secret.
+
+    Returns:
+        str: The OSD encryption secret if found, otherwise None.
+    """
+
+    secret_key = f"rook-ceph-osd-encryption-key-{device_handle}"
+
+    # Construct the Vault command
+    cmd = f"vault kv get -format=json {vault_backend_path}/{secret_key}"
+
+    try:
+        # Execute the command and capture the output
+        logger.info(f"Getting OSD secrets from vault : {cmd}")
+        out = subprocess.check_output(
+            shlex.split(cmd), stderr=subprocess.STDOUT, text=True
+        )
+
+        # Parse the JSON response
+        json_out = json.loads(out)
+
+        # Retrieve the secret
+        data_section = json_out.get("data", {})
+        secret = data_section.get("data", {}).get(
+            secret_key, data_section.get(secret_key)
+        )
+
+        if not secret:
+            logger.error(f"Secret for key '{secret_key}' not found in Vault response.")
```

Comment:
why not to throw the exception?

Reply:
Added Exception when secret is not found in the vault